### ISSUE TITLE
Parsed code-blocks might contain multiple TEXT and CONTENT nodes

### ISSUE THREAD

**Issue by svetlyak40wt** [CommonDoc/scriba#15]
I just want to clarify if this is an expected and desired behaviour.

Personally I'm expecting CODE-BLOCK nodes always have only a single TEXT-NODE child node (or probably even just keep it's content in a slot instead of having CHILDREN slot.

Here is a demo of a simple case, when parsing will lead to a multiple nodes. Pay attention, however STRONG markup does not parsed:

```lisp
CL-USER> (common-doc.format:parse-document
          (make-instance 'scriba:scriba)
          "@begin(code)
*Some*
(code)
block
@end(code)")
#<COMMON-DOC:DOCUMENT "">

CL-USER> (reblocks-text-editor/utils/common-doc:print-tree *)
#<CODE-BLOCK children: TEXT-NODE, CONTENT-NODE, TEXT-NODE>
  #<TEXT-NODE text: *Some*
>
  #<CONTENT-NODE children: TEXT-NODE, TEXT-NODE, TEXT-NODE>
    #<TEXT-NODE text: (>
    #<TEXT-NODE text: code>
    #<TEXT-NODE text: )>
  #<TEXT-NODE text:
block
>
```

**Comment by christophejunke** [comment]
I stumbled upon the same strange behavior, it comes directly from the Esrap grammar rules and is part of the test suite, I think it is thus expected and code processors must be ready to handle more than one child node.

In fact code blocks don't process their content differently from others, at least during parsing. The two asterisks are Markdown syntax and not bold in Scriba if I am not mistaken.

**Comment by svetlyak40wt** [comment]
You are correct about two asterisks. Scriba's markup is parsed inside a code block :(

```
REBLOCKS-TEXT-EDITOR/HTML/IMAGE> (common-doc.format:parse-document
          (make-instance 'scriba:scriba)
          "@begin(code)
@i(Some)
(code)
block
@end(code)")
#<COMMON-DOC:DOCUMENT "">
REBLOCKS-TEXT-EDITOR/HTML/IMAGE> (reblocks-text-editor/utils/common-doc:print-tree *)
#<CODE-BLOCK children: ITALIC, TEXT-NODE, CONTENT-NODE, TEXT-NODE>
  #<ITALIC children: TEXT-NODE>
    #<TEXT-NODE text: Some>
  #<TEXT-NODE text:
>
  #<CONTENT-NODE children: TEXT-NODE, TEXT-NODE, TEXT-NODE>
    #<TEXT-NODE text: (>
    #<TEXT-NODE text: code>
    #<TEXT-NODE text: )>
  #<TEXT-NODE text:
block
>
#<COMMON-DOC:DOCUMENT "">
```

**Comment by svetlyak40wt** [comment]
Also, I didn't find a way to escape `@i(Some)` to prevent it's processing by Scriba parser. Scriba's documentation uses this workaround with a file inclusion:

```
This is a begin/end block:

@code[lang=scribe](@include[path=includes/begin-end.scr]())
```

**Comment by svetlyak40wt** [comment]
@eudoxia0 what do you think about this issue? Should the parser be fixed to not parse content of code blocks?

**Comment by christophejunke** [comment]
what if there was a different syntax for all tags that operate on a literal block of text, so that it would still not be specific to "code" (we could define custom blocks), maybe by doubling the @ symbol or using another one: =@@tag()= or =&tag()=

there's also the question of how to escape the closing parentheses, or provide a different delimiter, I don't know

**Comment by eudoxia0** [comment]
Sorry for the delay. I think code blocks should be allowed to have markup so you can e.g. italicize code, strike out code, that kind of thing. It sometimes bothers me that markdown doesn't let you do this.

Re-reading the source code it seems this is intended: `code-block` inherits from `content-node` which has a list of child nodes.

**Comment by svetlyak40wt** [comment]
For my taste, italization, colors and other markup of code-blocks content should be done when this code block is emitted in other format. For example, knowing the code's language, we might apply colorization and output HTML as multiple nested `<span>`, `<strong>` and `<i>` nodes.

The fact code-blocks currently might include multiple nodes, makes it hard to apply such colorization algorithm when we are emitting it's content or to do some other things with the content, like generating intermediate files in literate-programming style.

Anyway, I think it is good idea to document the current behaviour if it is intentional. And also to mention escaping rules. For example, what should I do to make parser parse correctly a code-block with example of Sciba document.
